backend-rust/src/indexer.rs

```diff
@@ -3846,6 +3856,45 @@ struct PreparedValidatorSuspension {
     block_height: i64,
 }
 
+/// Represents a payday block with its CCD mint rewards and the associated block
+/// height. The block was identified by observing the
+/// `SpecialTransactionOutcome::Mint` event.
+struct PreparedPayDayBlock {
+    block_height: i64,
+    rewards:      Vec<PreparedUpdateAccountBalance>,
```

Comment:
It seems there will only ever be a vector of one element here, so maybe just:
```suggestion
    mint_reward:      PreparedUpdateAccountBalance,
```